### PR TITLE
Enable specifying custom Recombee region (also updates a lot of packages and rewrites to Vite)

### PR DESCRIPTION
**Pull request by JiriLojda**
### Motivation

Which issue does this fix? Fixes #`issue number`

If no issue exists, what is the fix or new feature? Were there any reasons to fix/implement things that are not obvious?

### Checklist

- [ ] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [ ] Tests are passing
- [ ] Docs have been updated (if applicable)
- [ ] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### How to test

If manual testing is required, what are the steps?

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -23,12 +23,13 @@ jobs:
...
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
       - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v4
         with:
           node-version: ${{ matrix.node-version }}
       - run: npm ci
       - uses: dprint/check@v2.2
+      - run: npm run lint
```
Comment (IvanKiral) [inline .github/workflows/main.yml:33]: where format :/

File: README.md [hunk 2]
```diff
@@ -160,7 +160,7 @@ $ npm install
 $ netlify dev
 ```
 
-The custom element is created with [Create React App](https://create-react-app.dev/).
+The custom element is created with [Vite](https://vitejs.dev/).
```
Comment (IvanKiral) [inline README.md:165]: let's chage the screenshot for new webhooks 💪

File: package-lock.json [hunk 3]
```diff
@@ -9,48 +9,42 @@
...
-        "@testing-library/user-event": "^13.5.0",
-        "@types/jest": "^27.5.2",
-        "@types/node": "^16.11.56",
-        "@types/react": "^18.0.17",
-        "@types/react-dom": "^18.0.6",
-        "axios": "^0.27.2",
-        "dprint": "^0.36.1",
-        "eslint": "^8.41.0",
+        "@kontent-ai/delivery-sdk": "^14.8.0",
+        "@kontent-ai/webhook-helper": "^2.1.0",
```
Comment (IvanKiral) [inline package-lock.json]: new webhooks should be realeased soon. let's see if dependabot will announce it

File: src/functions/model/recombee-client.ts [hunk 4]
```diff
@@ -82,12 +87,12 @@ export default class RecombeeClient {
...
           const dataType = this.datatypeMap.get(element.type);
           return dataType
-            ? new Recombee.requests.AddItemProperty(element.codename, dataType)
+            ? new Recombee.requests.AddItemProperty(element.codename ?? "", dataType)
             : null;
         })
         .filter(notNull),
     ];
-    return this.client.send(new Recombee.requests.Batch(requests));
+    return this.client.send(new Recombee.requests.Batch(requests)).then(() => {});
```
Comment (IvanKiral) [inline src/functions/model/recombee-client.ts]: nit: those looks kind of fun :D still no at level of my logging in data-ops

**Review by IvanKiral — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (JiriLojda) [reply .github/workflows/main.yml:33]: The step above `dprint/check@v2.2` checks formatting.

File: package-lock.json (hunk 3 above)
Comment (JiriLojda) [reply package-lock.json]: Dependabot only announces fixes for security issues. I don't believe that this update will be marked as such. 😄
But I updated the package anyway.

**Review by IvanKiral — APPROVED** [review]

**Pull request merged**